### ISSUE TITLE
Job with n(ext) runner crashes with directories as references

### ISSUE THREAD

**Issue by clebergnu** [avocado-framework/avocado#4094]
The following behavior can be seen on 8dc59da6d29166245cf88a0584eaa09e9acce383:

```
$ avocado run --test-runner=nrunner selftests/unit/
Avocado crashed unexpectedly: Could not resolve references: selftests/unit/
You can find details in /home/cleber/avocado/data/crashes/avocado-traceback-2020-08-18_22:23:44-j4pw89zs.log
```

And the crash log reveals that:

```
Avocado crashed:
Traceback (most recent call last):
  File "/home/cleber/.local/bin/avocado", line 11, in <module>
    load_entry_point('avocado-framework', 'console_scripts', 'avocado')()
  File "/home/cleber/src/avocado/avocado/avocado/core/main.py", line 76, in main
    return app.run()
  File "/home/cleber/src/avocado/avocado/avocado/core/app.py", line 112, in run
    return method(self.parser.config)
  File "/home/cleber/src/avocado/avocado/avocado/plugins/run.py", line 298, in run
    suite = TestSuite.from_config(config, name='suite01')
  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 186, in from_config
    suite = cls._from_config_with_resolver(config, name)
  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 84, in _from_config_with_resolver
    resolutions = resolve(references, ignore_missing=ignore_missing)
  File "/home/cleber/src/avocado/avocado/avocado/core/resolver.py", line 203, in resolve
    raise JobTestSuiteReferenceResolutionError(msg)
avocado.core.exceptions.JobTestSuiteReferenceResolutionError: Could not resolve references: selftests/unit/
```

The indented behavior is:

* No crashes
* Job may abort and not run if there are no resolved references
* Directories themselves should be ignored on missing references

**Comment by clebergnu** [comment]
Even with #4095, this remains an open issue.  Reproducer is:

```
$ avocado run --test-runner=nrunner /bin/true /dev/null
Avocado crashed unexpectedly: Could not resolve references: /dev/null
You can find details in /home/cleber/avocado/data/crashes/avocado-traceback-2020-08-18_22:42:12-dd21mi0u.log
```

The `JobTestSuiteReferenceResolutionError` needs to be properly handled, and possibly the test suite needs to have a flag for this situation.

**Comment by clebergnu** [comment]
I've pused 1b77348051c3d834b8134aded5a6dacba7426936 to #4095, and it seems to do the trick.

**Comment by clebergnu** [comment]
Solved by #4095
